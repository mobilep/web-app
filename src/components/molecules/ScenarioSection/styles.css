.ScenarioSection {
	padding: var(--scenario-section-vertical-padding) 0;
	border-top: var(--lightBorder);
}

.ScenarioSection-header {
	font-size: 12px;
	color: var(--grey3);
	display: flex;
	justify-content: space-between;
}

.ScenarioSection-content {
	width: 100%;
	margin-top: 12px;
	flex-grow: 1;
}

@media (max-width: 768px) {
	.ScenarioSection {
		--scenario-section-vertical-padding: 15px;
	}
}

.ScenarioSection-wrapperContent {
	display: flex;
	max-height: 500px;
}

.ScenarioSection-errorContent {
	width: 40px;
}
