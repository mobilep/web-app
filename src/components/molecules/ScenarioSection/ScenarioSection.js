import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { InputError } from '../../molecules';

import './styles.css';

export default class ScenarioSection extends Component {

	static propTypes = {
		children: PropTypes.any,
		className: PropTypes.string,
		counter: PropTypes.number,
		errorText: PropTypes.string,
		header: PropTypes.string,
		showValidationError: PropTypes.bool,
	}

	getClassName (className) {
		return classNames('ScenarioSection', className);
	}

	render () {
		const {
			className,
			header,
			counter,
			children,
			showValidationError,
			errorText,
		} = this.props;
		const counterText = counter ? `(${counter})` : null;

		return (
			<div className={this.getClassName(className)}>
				<div className='ScenarioSection-header'>
					{header} {counterText}
				</div>
				<div className='ScenarioSection-wrapperContent'>
					<div className='ScenarioSection-content'>
						{children}
					</div>
					{showValidationError &&
						<div className='ScenarioSection-errorContent'>
							<InputError popUpPosition='topRightPopUp' color='red' messageText={errorText} />
						</div>}
				</div>
			</div>
		);
	}

}
