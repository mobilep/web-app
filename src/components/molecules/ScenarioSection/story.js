import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { text, boolean, number } from '@storybook/addon-knobs/react';
import ScenarioSection from './ScenarioSection';
import { AppendableList } from '../../molecules';

const DEFAULT_ITEMS = [
	{ content: 'Some item' },
	{ content: 'Another item' },
	{ content: 'Third item' },
];

storiesOf('2. Molecules / ScenarioSection', module)
	.add('default',
		() => (<div>
			<ScenarioSection header={text('Text', 'steps')} counter={number('Counter', 1)}>
				{boolean('List', false) && <AppendableList
					items={DEFAULT_ITEMS}
					onAddItem={action('Clicked')}
					type='step' />}
			</ScenarioSection>
		</div>));
