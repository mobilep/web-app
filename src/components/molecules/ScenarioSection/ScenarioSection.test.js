import React from 'react';
import { shallow } from 'enzyme';

import ScenarioSection from './ScenarioSection';

describe('Molecule - ScenarioSection', () => {

	it('Should render', () => {
		const wrapper = shallow(
			<ScenarioSection
				counter={5}
				header='Header'
			>
				Section content
			</ScenarioSection>
		);
		expect(wrapper).toMatchSnapshot();
	});

});
