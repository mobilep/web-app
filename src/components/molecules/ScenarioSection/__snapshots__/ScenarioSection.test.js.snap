// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecule - ScenarioSection Should render 1`] = `
<div
  className="ScenarioSection"
>
  <div
    className="ScenarioSection-header"
  >
    Header
     
    (5)
  </div>
  <div
    className="ScenarioSection-wrapperContent"
  >
    <div
      className="ScenarioSection-content"
    >
      Section content
    </div>
  </div>
</div>
`;
